BlockchainSdk/Blockchains/Sui/SuiTransactionBuilder.swift

```diff
@@ -0,0 +1,132 @@
+//
+// SuiTransactionBuilder.swift
+// BlockchainSdk
+//
+// Created by Sergei Iakovlev on 30.08.2024
+// Copyright © 2024 Tangem AG. All rights reserved.
+//
+
+import Foundation
+import WalletCore
+import TangemSdk
+
+class SuiTransactionBuilder {
+    private let signer: any Address
+    private let publicKey: Wallet.PublicKey
+    private let decimalValue: Decimal
+    private var coins: [SuiCoinObject] = []
+    
+    init(publicKey: Wallet.PublicKey, decimalValue: Decimal) throws {
+        self.signer = try WalletCoreAddressService(coin: .sui).makeAddress(for: publicKey, with: .default)
+        self.publicKey = publicKey
+        self.decimalValue = decimalValue
+    }
+    
+    func update(coins: [SuiCoinObject]) {
+        self.coins = coins
+    }
+    
+    func buildForInspect(amount: Amount, destination: String, referenceGasPrice: Decimal) throws -> String {
+        let useCoins = coins
+        let totalAmount = coins.reduce(into: Decimal(0)) { partialResult, coin in
+            partialResult += coin.balance
+        }
+        
+        let decimalAmount = amount.value * decimalValue
+        let isSendMax = decimalAmount == totalAmount
+        
+        let availableBudget = isSendMax ? totalAmount - Decimal(1) : totalAmount - decimalAmount
+        let budget = min(availableBudget, SUIUtils.SuiGasBudgetMaxValue)
+        
+        let inputAmount = isSendMax ? totalAmount - budget : decimalAmount
+        
+        let input = try input(amount: inputAmount / decimalValue, destination: destination, fee: .init(gasPrice: referenceGasPrice, gasBudget: budget))
+        
+        let signatureMock = Data(repeating: 0x01, count: 64)
+        
+        let compiled = try TransactionCompiler.compileWithSignatures(coinType: .sui,
+                                                                     txInputData: input.serializedData(),
+                                                                     signatures: signatureMock.asDataVector(),
+                                                                     publicKeys: publicKey.blockchainKey.asDataVector())
```

Comment:
то же самое, так проще читается
```suggestion
        let compiled = try TransactionCompiler.compileWithSignatures(
            coinType: .sui,
            txInputData: input.serializedData(),
            signatures: signatureMock.asDataVector(),
            publicKeys: publicKey.blockchainKey.asDataVector()
        )
```